nixos/modules/installer/tools/nixos-version.sh

```diff
@@ -8,6 +8,11 @@ case "$1" in
   --hash|--revision)
     echo "@revision@"
     ;;
+  --json)
+    cat <<EOF
+{"nixosVersion": "@version@", "nixpkgsRevision": "@revision@", "configurationRevision": "@configurationRevision@"}
+EOF
```

Comment:
Couldn't we replace a more generic `@nixosVersionJSONPayload@` with a properly serialized JSON value, rather than embedding a couple ad-hoc properties in the flow of the script?